virt-v2v/pkg/customize/scripts/rhel/run/network_config_util.sh

```diff
@@ -46,6 +46,42 @@ extract_mac_ip() {
     fi
 }
 
+netplan_get_py() {
+    python -c "
+import os
+import yaml
+import sys
+
+netplan_dir = os.getenv('NETPLAN_DIR', '') + '/etc/netplan'
+args = sys.argv[1].split('.')
+
+def find_yaml_files(directory):
+    yaml_files = []
+    for root, dirs, files in os.walk(directory):
+        for file in files:
+            if file.endswith(('.yaml', '.yml')):
+                yaml_files.append(os.path.join(root, file))
+
+    return yaml_files
+
+def get_yaml_path(yaml_data, keys):
+    for key in keys:
+        if yaml_data is None:
+            return None
+        yaml_data = yaml_data.get(key)
+    return yaml_data
+
+yaml_files = find_yaml_files(netplan_dir)
+
+for yaml_file in yaml_files:
+    with open(yaml_file, 'r') as file:
+        yaml_data = yaml.safe_load(file)        
+        result = get_yaml_path(yaml_data, ['network'] + args)
+        if result is not None:
+           print(yaml.dump(result, default_flow_style=False))
```

Comment:
Do we know which python is used in the ubuntu 18?

Reply:
> Do we know which python is used in the ubuntu 18?

python3, on my ubuntu18 (new installed using iso) the python version was 3.6.9

>we would have problems with python2

yes, this script will not work on python2 

AFAIK (* not ubuntu expert)
18.04 - python3 become default python
17.10 - netplan become default network manager
( note: 17 is not LTS version, first LTS version to use netplan is 18.04 )

@cc @fabiand ^^

Reply:
Okok thanks for checking, sgtm